standard/attributes.md

```diff
@@ -1060,6 +1060,44 @@ Specifies that a nullable argument won’t be `null` when the method returns the
 >
 > *end example*
 
+### §enumerator-cancellation The EnumeratorCancellation attribute
+
+Specifies the parameter representing the `CancellationToken` for an asynchronous iterator (§15.15). The value of this parameter is a combination of two cancellation tokens: the token which was actually passed to the method, and the token passed to the `IAsyncEnumerable<T>.GetAsyncEnumerator(CancellationToken)` implementation on the generated iterator. The combined token will be canceled if either of the two source tokens are canceled.
+
+It is an error if the `EnumeratorCancellation` attribute is applied to more than one parameter. The compiler may produce a warning if the `EnumeratorCancellation` attribute is applied to a parameter of a different type than `CancellationToken` or on a method that isn't an asynchronous iterator (§15.15) which returns an asynchronous enumerable interface ([§15.15.3](classes.md#15153-enumerable-interfaces)) rather than an asynchronous enumerator interface ([§15.15.2](classes.md#15152-enumerator-interfaces)). If no attributes have this parameter, the iterator won't have access to the `CancellationToken` argument for `GetAsyncEnumerator`.
+
+> *Example*: The method `GetStringsAsync()` is an asynchronous iterator. Before doing any work to retrieve the next value, it checks the cancellation token to determine if the iteration should be canceled. If cancellation is requested, no further action is taken.
+>
+> <!-- Example: {template:"code-in-class-lib", name:"AsyncEnumeratorCancellation"} -->
+> ```csharp
+> static async IAsyncEnumerable<object> GetStringsAsync([EnumeratorCancellation] CancellationToken token)
+> {
+>     for (int i = 0; i < 10; i++)
+>     {
+>         if (token.IsCancellationRequested)
+>         {
+>             yield break; // Exit if cancellation is requested
+>         }
+>         await Task.Delay(100);
+>         yield return i.ToString();
+>     }
+> }
+> ```
+>
+> The `EnumerationCancellation` attribute instructs the compiler that the associated parameter is used for cancellation. The cancellation token passed to `GetAsyncEnumerator` is combined with `sourceOne.Token` into a single combined token and substituted for the argument to this parameter. In the following example, one token is passed to the `GetStringsAsync` method, and another is passed to the `ToListAsync` extension method. The compiler can create a combined token that requests cancellation when either token has requested cancellation. The async iterator then cancels the operation based on the state of the combined token.
```

Comment:
```suggestion
> The `EnumerationCancellation` attribute instructs the compiler that the associated parameter is used for cancellation. In the following example, one token is passed to the `GetStringsAsync` method, and another is passed to the `ToListAsync` extension method. The cancellation token passed to `GetAsyncEnumerator` is combined with `sourceOne.Token` into a single combined token and substituted for the argument to this parameter. The compiler can create a combined token that requests cancellation when either token has requested cancellation. The async iterator then cancels the operation based on the state of the combined token.
```

(Just reordering the sentences, otherwise we're referring to `sourceOne.Token` without introducing the example code that contains it.)

Reply:
This is in an example so maybe this is a quibble, but “The compiler can” is surely wrong. If token combination is in the Standard, as I expect it is, then in normative text the implementation *must*/*shall* combine the tokens, in informative descriptive text maybe the implementation *will* combine?